weldx/util/xarray.py

```diff
@@ -529,13 +530,47 @@ def xr_check_coords(dax: xr.DataArray, ref: dict) -> bool:
                 raise DimensionalityError(
                     units,
                     check["dimensionality"],
-                    f"\nDimensionality mismatch in coordinate '{key}'\n"
+                    extra_msg=f"\nDimensionality mismatch in coordinate '{key}'\n"
                     f"Coordinate has unit '{units}', expected '{dim}'",
                 )
 
     return True
 
 
+def xr_check_dimensionality(da: xr.DataArray, units_ref: Union[str, pint.Unit]):
+    """Check if a `DataArray`s dimensionality matches the one of a reference unit.
+
+    Parameters
+    ----------
+    da:
+        The data array that should be checked.
+    units_ref:
+        The reference unit
+
+    Raises
+    ------
+    pint.DimensionalityError:
+        The error is raised if the check fails
+
+    """
+    if units_ref is None:
+        return
+    if not isinstance(units_ref, pint.Unit):
+        units_ref = U_(units_ref)
```

Comment:
I think be default it's ok to simply cast `units_ref = U_(units_ref)` at the beginning without type checking checking